engine/docker/docker.go

```diff
@@ -183,24 +184,44 @@ func (e *Docker) Destroy(ctx context.Context, pipelineConfig *spec.PipelineConfi
 
 // Run runs the pipeline step.
 func (e *Docker) Run(ctx context.Context, pipelineConfig *spec.PipelineConfig, step *spec.Step,
-	output io.Writer) (*runtime.State, error) {
+	output io.Writer, isDrone bool) (*runtime.State, error) {
 	// create the container
 	err := e.create(ctx, pipelineConfig, step, output)
 	if err != nil {
 		return nil, errors.TrimExtraInfo(err)
 	}
+	// start the execution in go routine if it's a detach step and not drone
+	if !isDrone && step.Detach {
+		go func() {
+			ctxBg := context.Background()
+			var cancel context.CancelFunc
+			if deadline, ok := ctx.Deadline(); ok {
```

Comment:
Could we not use the same context which is passed in? Does that get passed around in other steps as well?